clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -310,21 +310,31 @@ public KafkaProducer(Properties properties) {
      *                         be called in the producer when the serializer is passed in directly.
      */
     public KafkaProducer(Properties properties, Serializer<K> keySerializer, Serializer<V> valueSerializer) {
-        this(propsToMap(properties), keySerializer, valueSerializer, null, null, null,
+        this(new ProducerConfig(ProducerConfig.addSerializerToConfig(properties, keySerializer, valueSerializer)), keySerializer,
+                valueSerializer, null, null, null,
                 Time.SYSTEM);
     }
 
     // visible for testing
-    @SuppressWarnings("unchecked")
     KafkaProducer(Map<String, Object> configs,
```

Comment:
this constructor is used by KafkaProducerTest only. I keep this method to minimize this PR